pkg/controller/controller_instance.go

```diff
@@ -1696,6 +1844,7 @@ func (c *controller) processTerminalProvisionFailure(instance *v1beta1.ServiceIn
 	if failedCond == nil {
 		return fmt.Errorf("failedCond must not be nil")
 	}
+	c.removeInstanceFromRetryMap(instance)
 	return c.processProvisionFailure(instance, readyCond, failedCond, shouldMitigateOrphan)
```

Comment:
Here and in other places: there is a chance that we will fail to update the status inside `processProvisionFailure`, but we have already removed instance from retry map. Which means that operation will be retried without backoff.
Probably fine, given that the probability of this seems low, and it won't break anything.